src/services/ui/src/features/package-actions/ActionWrapper.tsx

```diff
@@ -0,0 +1,56 @@
+import { FormProvider, useForm } from "react-hook-form";
+import { Outlet, useParams } from "react-router-dom";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { AnyZodObject } from "zod";
+import { BreadCrumbs, SimplePageContainer } from "@/components";
+import { detailsAndActionsCrumbs } from "@/pages/actions/actions-breadcrumbs";
+import { Action } from "shared-types";
+import { issueRaiSchema } from "./IssueRai";
+import { withdrawRaiSchema } from "./WithdrawRai";
+import { toggleRaiResponseWithdrawSchema } from "./ToggleRaiResponseWithdraw";
+import { withdrawPackageSchema } from "./WithdrawPackage";
+import { respondToRaiSchema } from "./RespondToRai";
+
+const schemas = {
+  "issue-rai": issueRaiSchema,
+  "withdraw-rai": withdrawRaiSchema,
+  "enable-rai-withdraw": toggleRaiResponseWithdrawSchema,
+  "disable-rai-withdraw": toggleRaiResponseWithdrawSchema,
+  "withdraw-package": withdrawPackageSchema,
+  "respond-to-rai": respondToRaiSchema,
+} satisfies Record<string, AnyZodObject>;
+type SchemaKeys = keyof typeof schemas;
+
+const actions: Record<SchemaKeys, Action> = {
+  "issue-rai": Action.ISSUE_RAI,
+  "disable-rai-withdraw": Action.DISABLE_RAI_WITHDRAW,
+  "enable-rai-withdraw": Action.ENABLE_RAI_WITHDRAW,
+  "respond-to-rai": Action.RESPOND_TO_RAI,
+  "withdraw-package": Action.WITHDRAW_PACKAGE,
+  "withdraw-rai": Action.WITHDRAW_RAI,
+};
+
+export const ActionWrapper = () => {
+  const packageActionType = window.location.href
+    .split("/")
+    .at(-1) as SchemaKeys;
```

Comment:
That's a great question. I tried the useParams() hook initially. But it would not pull the param out of it. Maybe a weird edge case of the router

Reply:
Oooh. I'll check that out!!

Reply:
Bumping, doesn't look like this was remedied yet, but agree with Ben, let's use this with useParams. Paul's implementation of useParams has some type safety built in, check out the various action forms for examples of its use.

Reply:
Yes. This is all great convo and suggestions. Maybe I can look into adding some logic to that custom hook in a way that it can pull out things and type check them still. Maybe we can bundle the useLocation hook into Paul's custom one if that makes sense?